packages/fluentui/react-northstar/src/utils/positioner/usePopper.ts

```diff
@@ -0,0 +1,408 @@
+         * default. i.e. popper is still rendered fully in the viewport even if anchor element is no longer in the
+         * viewport.
+         */
+        unstable_disableTether && {
+          name: 'preventOverflow',
+          options: { altAxis: unstable_disableTether === 'all', tether: false },
+        },
+
+        flipBoundary && {
+          name: 'flip',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, flipBoundary),
+          },
+        },
+        overflowBoundary && {
+          name: 'preventOverflow',
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+        },
+
+        {
+          name: 'onUpdate',
+          enabled: true,
+          phase: 'afterWrite' as PopperJs.ModifierPhases,
+          fn: handleStateUpdate,
+        },
+
+        autoSize && {
+          // Similar code as popper-maxsize-modifier: https://github.com/atomiks/popper.js/blob/master/src/modifiers/maxSize.js
+          // popper-maxsize-modifier only calculates the max sizes. This modifier applies the max sizes when overflow is detected
+          name: 'applyMaxSize',
+          enabled: true,
+          phase: 'beforeWrite' as PopperJs.ModifierPhases,
+          requiresIfExists: ['offset', 'preventOverflow', 'flip'],
+          options: {
+            altBoundary: true,
+            boundary: getBoundary(container, overflowBoundary),
+          },
+          fn({ state, options }: PopperJs.ModifierArguments<{}>) {
+            const overflow = PopperJs.detectOverflow(state, options);
+            const { x, y } = state.modifiersData.preventOverflow || { x: 0, y: 0 };
+            const { width, height } = state.rects.popper;
+            const [basePlacement] = state.placement.split('-');
+
+            const widthProp: keyof PopperJs.SideObject = basePlacement === 'left' ? 'left' : 'right';
+            const heightProp: keyof PopperJs.SideObject = basePlacement === 'top' ? 'top' : 'bottom';
+
+            if (overflow[widthProp] > 0 && (autoSize === true || autoSize === 'width')) {
+              state.styles.popper.maxWidth = `${width - overflow[widthProp] - x}px`;
+            }
+            if (overflow[heightProp] > 0 && (autoSize === true || autoSize === 'height')) {
+              state.styles.popper.maxHeight = `${height - overflow[heightProp] - y}px`;
+            }
+          },
+        },
+
+        /**
+         * This modifier is necessary in order to render the pointer. Refs are resolved in effects, so it can't be
+         * placed under computed modifiers. Deep merge is not required as this modifier has only these properties.
+         */
+        {
+          name: 'arrow',
+          enabled: !!arrow,
+          options: { element: arrow },
+        },
+      ].filter(Boolean);
+
+      const options: PopperJs.Options = {
```

Comment:
There are multiple scopes of `options` in this hook

```suggestion
      const popperOptions: PopperJs.Options = {
```